src/www/comm.js

```diff
@@ -0,0 +1,206 @@
+
+/**
+ * @description Formation website server communication module.
+ * @module comm
+ * @see module:comm
+ * @author mads256h
+ */
+
+(function() {
+    // Requests
+
+    /**
+     * @public
+     * @memberof module:comm
+     * @summary Send a login request
+     * @description This request sets a session cookie in the browser.
+     * This cookie is required in the all the other requests.
+     * @param {string} username The name of the student to log in as
+     * @returns {Promise<undefined>} The response
+     * @throws {Error} The status of the response is not "OK"
+     */
+    function login(username) {
+        return postRequest("login", {username});
+    }
+
+    /**
+     * @public
+     * @memberof module:comm
+     * @summary Get student information about the logged in student
+     * @description This request requires that a valid session cookie has been set
+     * @returns {Promise<module:group~Student>} A promise with the student
+     * @throws {Error} The status of the response is not "OK"
+     */
+    function me() {
+        return getRequest("me");
+    }
+
+    /**
+     * @public
+     * @memberof module:comm
+     * @summary Get group information about the logged in student
+     * @description This request requires that a valid session cookie has been set
+     * @returns {Promise<module:group~Group>} A promise with the students group
+     * @throws {Error} The status of the response is not "OK"
+     */
+    function mygroup() {
+        return getRequest("mygroup");
+    }
+
+    /**
+     * @typedef {object} RankedGroup
```

Comment:
Den her doc virker misplaced/outdated

Reply:
Nej den bliver brugt lige neden under `* @returns {Promise<RankedGroup>} A promise with the ranked groups`

Reply:
Oops ja det skal være et array

Reply:
Fixed